### ISSUE TITLE
Mypy plugins does not work with lunarvim

### ISSUE THREAD

**Issue by autra** [LunarVim/LunarVim#4502]
### Problem description

When a plugin is activated in a project in mypy.ini, the mypy linter fails to start

### LunarVim version

master-4ffa7c45

### Neovim version (>= 0.9.1)

0.9.1

### Operating system/version

Ubuntu/22.04

### Affected language servers

pylsp

### Steps to reproduce

1. create a project: `mkdir -p test_case/src`
2.  `git init` in `test_case` (not sure if necessary, but just in case)
 3. put the following in mypy.ini
    ```ini
	[mypy]
	files = src/, tests/
	plugins = pydantic.mypy
	```
4. edit a python file in src: `lvim src/test.py`

### Actual behavior

lunarvim starts with this error:

```
[null-ls] failed to run generator: ...t/none-ls.nvim/lua/null-ls/helpers/generator_factory.lua:228: error in generato
r output: /home/augustin/workspace/berger_levrault/berger_levrault_carl_3d/mypy.ini:3:1: error: Error importing plugi
n "pydantic.mypy": No module named 'pydantic'  [misc]
```

### Expected behavior

LunarVim starts without errors, and typing issues are linted by mypy.

### support info

LspInfo:
```console
 Press q or <Esc> to close this window. Press <Tab> to view server doc.
 
 Language client log: /home/augustin/.local/state/lvim/lsp.log
 Detected filetype:   python
 
 2 client(s) attached to this buffer: 
 
 Client: pylsp (id: 1, bufnr: [1])
 	filetypes:       python
 	autostart:       true
 	root directory:  /home/augustin/repos/test_bug
 	cmd:             /home/augustin/.local/share/lvim/mason/bin/pylsp
 
 Client: null-ls (id: 2, bufnr: [1])
 	filetypes:       typescript, typescriptreact, javascriptreact, vue, javascript, python, sql
 	autostart:       false
 	root directory:  /home/augustin/repos/test_bug
 	cmd:             <function>
 
 Configured servers list: sqls, dockerls, bashls, tsserver, volar, eslint, pylsp, ansiblels
```
LvimInfo
```console

    __                          _    ___         
   / /   __  ______  ____ _____| |  / (_)___ ___ 
  / /   / / / / __ \/ __ `/ ___/ | / / / __ `__ \
 / /___/ /_/ / / / / /_/ / /   | |/ / / / / / / /
/_____/\__,_/_/ /_/\__,_/_/    |___/_/_/ /_/ /_/ 


Buffer info
* filetype:                python
* bufnr:                   1
* treesitter status:       active

Active client(s)
* name:                      pylsp
* id:                        1
* supported filetype(s):     python
* attached buffers:          1
* root_dir pattern:          1
* capabilities:              renameProvider                  | documentHighlightProvider | foldingRangeProvider
                             referencesProvider              | hoverProvider             | codeActionProvider
                             documentRangeFormattingProvider | definitionProvider
                             documentFormattingProvider      | documentSymbolProvider

Automatic LSP info
* Skipped servers: [angularls, ansiblels, antlersls, ast_grep, azure_pipelines_ls, biome, cssmodules_ls, custom_elements_ls, denols, docker_compose_language_service, elp, ember, emmet_language_server, emmet_ls, eslint, glint, golangci_lint_ls, gradle_ls, graphql, htmx, java_language_server, jedi_language_server, ltex, mdx_analyzer, neocmake, nim_langserver, omnisharp, phpactor, psalm, pylsp, pylyzer, pyre, quick_lint_js, reason_ls, rnix, rome, rubocop, ruby_ls, ruff_lsp, solang, solc, solidity_ls_nomicfoundation, sorbet, sourcery, spectral, sqlls, sqls, standardrb, stimulus_ls, stylelint_lsp, svlangserver, tflint, unocss, verible, v_analyzer, vtsls, vuels]

Formatters info
* Active: black  
* Supported: [autoflake, autopep8, black, blackd, blue, docformatter, isort, isortd, pyflyby, pyink, reorder_python_imports, ruff, ruff_format, usort, yapf]

Linters info
* Active: 
* Supported: [bandit, flake8, mypy, pycodestyle, pydocstyle, pylama, pylint, pyproject_flake8, ruff, semgrep, vulture]

Code actions info
* Active: 



### logs

_No response_

### Screenshots

_No response_

**Comment by xidiot** [comment]
@autra , I have the same issue (the 2nd one with "index out of range") like you after installing flake8 (`:MansoInstall flake8`).  A workaround (maybe not breaking anything?) is to comment the following piece of code (line 70 to 73) in file `~/.local/share/lunarvim/site/pack/lazy/opt/none-ls.nvim/lua/null-ls/helpers/diagnostics.lua`:

`if entries["col"] ~= nil and content_line ~= nil then`
        `local byte_index_col = vim.str_byteindex(content_line, tonumber(entries["col"]))`
        `entries["col"] = tostring(byte_index_col)`
`end`

**Comment by chenxia-han** [comment]
I encountered the same **index out of range** issue recently, and it can be easily reproduced with the following steps:
```bash
git clone git@github.com:mlsyscourse/homework1.git
cd homework1
lvim logistic_regression.py # You will see the error message at the bottom
```
The root cause of this error is related to mypy, which outputs error messages not only for the current file but also for imported files. Consequently, this leads to an error when null-ls attempts to parse the error messages from these additional files.

I was able to resolve this by modifying the LunarVim configuration to include the `--follow-imports=silent` argument for mypy. This change ensures that error messages from other files are suppressed.
